### ISSUE TITLE
TypeScript errors (TS2416) when compiling an app in strict mode due to mismatch in interface implementations

### ISSUE THREAD

**Issue by StefanNedelchev** [cornerstonejs/cornerstone3D-beta#504]
I have imported `@cornerstonejs/core` in an Angular app which uses strict mode in tsconfig and I implemented a simple code that should load a single image in a viewport. The build fails due to some TypeScript errors in the cornerstone type definitions:

![image](https://user-images.githubusercontent.com/15238282/226999637-572c603a-f770-4549-9881-00b38f90c16d.png)

The screenshot includes only part of the errors but the others suffer from similar symptoms - some methods inside class declarations don't exactly match the methods from the interfaces they implement.

An example with `StackViewport` where the return type of `getFrameOfReferenceUID()` includes `undefined` but in the interface `IStackViewport` and `IViewport` the return type is strictly a `string`. Also I'm not sure why `IStackViewport` duplicates so many things from `IViewport` given the fact it already extends `IViewport` (the same thing is noticed in other interfaces).
![image](https://user-images.githubusercontent.com/15238282/227000389-25930ed4-6bdb-40a7-9dbf-ce757a69b6c1.png)

**Comment by StefanNedelchev** [comment]
@sedghi Thanks, I might try to create PRs for each of the issues I've raised

**Comment by sedghi** [comment]
I will try to create a branch that experiment with fixing those 657 errors

**Comment by VijayanRamachandran** [comment]
Hi @sedghi @StefanNedelchev ,

i am also facing same issue in Angular when i use latest version of Cornerstone3D 

![image](https://github.com/cornerstonejs/cornerstone3D/assets/25306014/65c5cdce-3e94-4be8-a9c2-16ad01b2b75a)

Any suggestion about temporary workaround to solve it or already any permanent fix to it which you could share please ?

Regards,
Vijayan Ramachandran

**Comment by StefanNedelchev** [comment]
> Hi @sedghi @StefanNedelchev ,
> 
> i am also facing same issue in Angular when i use latest version of Cornerstone3D
> 
> ![image](https://user-images.githubusercontent.com/25306014/270566746-65c5cdce-3e94-4be8-a9c2-16ad01b2b75a.png)
> 
> Any suggestion about temporary workaround to solve it or already any permanent fix to it which you could share please ?
> 
> Regards, Vijayan Ramachandran

Hi. Unfortunately I decided to give up trying until Cornerstone3D officially goes in strict TS mode (which is difficult to prioritize when they focus on bug fixing and features). As for a temporary solution, unfortunately I'm not aware of other way than setting the `strict` option to `false` in your tsconfig. The downside is that it would disable it entirely in the project but I think it's impossible to tweak the config in a way to ignore strict type checking for `node_modules`

**Comment by sedghi** [comment]
Yeah as Stefan mentioned, unfortunately it is not a priority at the moment, although I'm more than happy to review a PR if someone is trying to fix our types

**Comment by VijayanRamachandran** [comment]
Hi @sedghi @StefanNedelchev ,

Thanks for the quick response..

for now i am proceeding with below configurations in tsconfig.json 

"types": ["node"],
"noImplicitAny": false,
"strictNullChecks":false,
"skipLibCheck": true,

Regards,
Vijayan Ramachandran
